### PR TITLE
Use QTokenizer to properly parse netrc

### PR DESCRIPTION
**Pull request by danimo**
Addresses #4691

### CODE REVIEW

**Comment by danimo** [comment]
@ogoffart pls

File: src/3rdparty/qtokenizer/qtokenizer.h [hunk 1]
```diff
@@ -0,0 +1,264 @@
+/****************************************************************************
+**
+** Copyright (C) 2014 Daniel Molkentin <daniel@molkentin.de>
+** Contact: http://www.qt-project.org/legal
```
Comment (ogoffart) [inline src/3rdparty/qtokenizer/qtokenizer.h]: why qt-project?

Comment (danimo) [inline src/3rdparty/qtokenizer/qtokenizer.h]: It's supposed to end up in Qt eventually. I can change the header if needed.

File: src/3rdparty/qtokenizer/qtokenizer.h [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+};
+
+typedef QTokenizer<QByteArray, QByteArray::const_iterator> QByteArrayTokenizer;
+typedef QTokenizer<std::string, std::string::const_iterator> StringTokenizer;
+typedef QTokenizer<std::wstring, std::wstring::const_iterator> WStringTokenizer;
+
+QT_END_NAMESPACE
+
+#endif // TOKENIZER_H
+
```
Comment (dragotin) [inline src/3rdparty/qtokenizer/qtokenizer.h]: No special marking of member variables (we use the prepended '_') and implementations in the header file - both not our style, but I assume that's cool in Qt?

File: src/3rdparty/qtokenizer/qtokenizer.h [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+       \param delimiters A string containing delimiters
+
+       \sa QStringTokenizer, QByteArrayTokenizer, StringTokenizer, WStringTokenizer
+     */
+    QTokenizer(const T& string, const T& delimiters) {
+        d.reset(new QTokenizerPrivate<T, const_iterator>(string, delimiters));
+    }
+
+    /*!
+       Whether or not to return delimiters ase tokens
```
Comment (dragotin) [inline src/3rdparty/qtokenizer/qtokenizer.h]: typo "ase"

**Comment by dragotin** [comment]
I think the Tokenizer class should accept Regexps at one point.

Otherwise :+1:

File: src/3rdparty/qtokenizer/qtokenizer.h (hunk 2 above)
Comment (danimo) [inline src/3rdparty/qtokenizer/qtokenizer.h]: Right :)

**Pull request merged**